static/app/views/ddm/focusArea.tsx

```diff
@@ -50,19 +50,29 @@ function isInRect(x: number, y: number, rect: DOMRect | undefined) {
   return x >= rect.left && x <= rect.right && y >= rect.top && y <= rect.bottom;
 }
 
-export function useFocusArea(
-  chartRef: RefObject<ReactEchartsRef>,
-  focusArea: FocusArea | null,
-  {widgetIndex, isDisabled = false, useFullYAxis = false}: UseFocusAreaOptions,
-  onAdd: (area: FocusArea) => void = () => {},
-  onRemove: () => void = () => {},
-  onZoom: (range: DateTimeObject) => void = () => {}
-) {
+type UseFocusAreaProps = {
+  chartRef: RefObject<ReactEchartsRef>;
+  focusArea: FocusArea | null;
+  opts: UseFocusAreaOptions;
+  onAdd?: (area: FocusArea) => void;
+  onDraw?: () => void;
+  onRemove?: () => void;
+  onZoom?: (range: DateTimeObject) => void;
+};
+
+export function useFocusArea({
+  chartRef,
+  focusArea,
+  opts: {widgetIndex, isDisabled, useFullYAxis},
+  onAdd = () => {},
```

Comment:
Careful - this is not an instance initializer and will result in a new function being initialized on each call. This will invalidate reference checks in hook dependencies (like on L94) and can result in an infinite loop.

Reply:
Thanks 👍 Fixed in [6b38ff8](https://github.com/getsentry/sentry/pull/63671/commits/6b38ff85992ad7a7d83c2d38e7f9ba83a40bc3ae)